alembic_migration/mysql/env.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2022 TOSIT.IO
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+
+from alembic import context
+from sqlalchemy import engine_from_config, pool
+
+from tdp.core.models.base_model import BaseModel
+
+# this is the Alembic Config object, which provides
+# access to the values within the .ini file in use.
+config = context.config
+
+# Model's MetaData object here for 'autogenerate' support.
+target_metadata = BaseModel.metadata
+
+
+def run_migrations_offline() -> None:
+    """Run migrations in 'offline' mode. Not implemented."""
+    raise NotImplementedError()
+
+
+def run_migrations_online() -> None:
+    """Run migrations in 'online' mode."""
+
+    engine_config = config.get_section(config.config_ini_section, {})
+
+    if not "sqlalchemy.url" in engine_config:
+        if not (database_dsn := os.environ.get("TDP_ALEMBIC_MYSQL_DSN")):
+            raise ValueError("TDP_ALEMBIC_MYSQL_DSN env var is missing")
+        engine_config["sqlalchemy.url"] = database_dsn
+
+    connectable = engine_from_config(
+        engine_config,
+        prefix="sqlalchemy.",
+        poolclass=pool.NullPool,
+    )
+
+    if connectable.dialect.name != "mysql":
+        raise ValueError("You are not connected to a MySQL or MariaDB database")
```

Comment:
We should provide some comments

```suggestion
    # Get the configuration based on the current section name
    engine_config = config.get_section(config.config_ini_section, {})

    # Override the database URL with the one from the environment if not set
    if not "sqlalchemy.url" in engine_config:
        if not (database_dsn := os.environ.get("TDP_ALEMBIC_MYSQL_DSN")):
            raise ValueError("TDP_ALEMBIC_MYSQL_DSN env var is missing")
        engine_config["sqlalchemy.url"] = database_dsn

    connectable = engine_from_config(
        engine_config,
        prefix="sqlalchemy.",
        poolclass=pool.NullPool,
    )

    # Ensure that the provided database url is of the proper dialect
    if connectable.dialect.name != "mysql":
        raise ValueError("You are not connected to a MySQL or MariaDB database")
```